arroyo-api/src/sources.rs

```diff
@@ -329,7 +344,7 @@ impl TryFrom<SchemaField> for SourceField {
     type Error = String;
 
     fn try_from(value: SchemaField) -> Result<Self, Self::Error> {
-        let typ = match value.typ {
+        let typ = match value.clone().typ {
```

Comment:
The clone is not necessary, just need to take a reference to value.typ instead of taking it by value

```suggestion
        let typ = match &value.typ {
```

Reply:
You would still need to clone that (as you try_into() is implemented on the owned type) but it's preferable to cloning the entire struct.